### ISSUE TITLE
Add custom pmd rule to sonar 5.4

### ISSUE THREAD

**Issue by emnvn** [jborgers/sonar-pmd#27]
Hello,

I m using Sonar 3.4 with PMD. I have written some pmd rule in java, then I add these rules to sonar 3.4, the result that it work well.

I have plan to use sonar 5.4, when I try to add my pmd rule to sonar 5.4, It can not load these rule, I can't see any error report too.

How can I use my custom pmd rule in sonar 5.4 ?

Thanks for helping!

**Comment by timwebster9** [comment]
Hi,

It's not clear to me whether the platform still supports PMD extensions.  Some Googling turned up this (although it looks like it might not have been resolved):

https://sourceforge.net/p/pmd/discussion/188193/thread/f52eec4e/

Have you tried this?

**Comment by emnvn** [comment]
I have tried it but It can't resolve  my problem. I don't know whether it is supported or not. If Sonar don't support custom rules why it still support pmd plugin ?

**Comment by dgeissl** [comment]
I have the same problem with SonarQube 4.5.6 and sonar-pmd-plugin 2.4.1/2.6. Didn't see that in the first place as we have only a small amount of custom rules and they still show up in the quality profile (as we used to embed them in a sonar-qube-plugin), but none of them is actually working.

I've tried to go back to sonar-pmd-plugin 2.3 as back then the <SONAR_HOME>/extensions/rules/pmd folder seems to have been scanned for extension xmls (altough there seems to be no schema around how to write them...) and with the help of http://sonarqube-archive.15.x6.nabble.com/PMD-custom-rule-problem-td3196265.html i got SonarQube to pick the rules up (with 2.3), but then pmd can't find the rules in the rules.jar, as it is not in the classpath.

Any ideas? I am perfectly fine with a extension to the sonar-pmd-plugin, as this works way better then having to provide a xml and jar in a separate folder, but this works only to the point that i can assign rules to quality profiles (so the internal extension.xml is read). During the analysis this plugin is not executed and the rules do not show up in the pmd rules permalink.
